param/parameterized.py

```diff
@@ -2698,6 +2713,32 @@ def pprint(self, imports=None, prefix="\n    ",unknown_value='<?>',
 
 
 
+class default_label_formatter(ParameterizedFunction):
```

Comment:
We usually use function-style names for ParameterizedFunctions, to make it more natural for people to call them like functions, but here it might be more natural to have a name like `LabelFormatter`, so that users can do `LabelFormatter.replace_underscores=False`.  `default_label_formatter.replace_underscores=False` is ok, but seems less obvious to me in what it does.  But if you disagree, feel fine to ignore this comment.

Reply:
```
LabelFormatter.replace_underscores=False
```

This would mean mutating the formatter in use (the actual parameterized function, not instances of it). I am ok with this approach, but right now, because of this line:

```
label_formatter = default_label_formatter.instance()
```

it won't currently work. The question is whether it is better to have a single label formatter function you mutate for customization, or a way to set any label formatter function you want, e.g a lambda function (what I've done in this PR). What *would* currently work in the style you suggest is this:

```
label_formatter.replace_underscores=False
```

Which mutates the default `default_label_formatter` instance...

Reply:
I agree that the latter would work.  But what does `.instance()` achieve, compared to simply doing `label_formatter=default_label_formatter`?  Was it to do with pickling?  I can't recall..

Reply:
There is no particularly strong reason to use `.instance` (I didn't think it would harm anything and would isolate the original parameterized function from mutations on `label_formatter` if it were used elsewhere...which is unlikely) so I removed it in f8032bb